pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -116,7 +116,9 @@ func identityMatches(set *apps.StatefulSet, pod *v1.Pod) bool {
 	return ordinal >= 0 &&
 		set.Name == parent &&
 		pod.Name == getPodName(set, ordinal) &&
-		pod.Namespace == set.Namespace
+		pod.Namespace == set.Namespace &&
+		pod.Spec.Hostname != "" &&
+		pod.Spec.Subdomain != ""
```

Comment:
I just had a thought:

ServiceName is allowed to be empty in 1.7+ [1], so we should do something like:

```go
(pod.Spec.Subdomain != "" || set.ServiceName == "")
```

Otherwise we will continually send Pod updates for any StatefulSet with an empty ServiceName.

[1] The reason ServiceName can't be empty in <1.7 is because the annotation gets added anyway, and then the Pod fails validation. In 1.7+ we no longer set the annotation, and the field has no distinction between `""` and unset.